pyomo/core/base/block.py

```diff
@@ -1006,6 +1006,9 @@ def del_component(self, name_or_object):
         # if name not in self._decl:
         #    return
 
+        # Call deconstruction routine in case there is any cleanup
+        obj.deconstruct()
```

Comment:
I created one. It's mostly because constructing an Arc adds itself to lists located on the member ports, and I wanted a way to clean up those lists if the user deletes an Arc from the model so that it doesn't linger in there. As far as I know Arc is the only component that modifies other components when it's constructed so that's why the need for this is new.

Reply:
I believe that if you add a var to a port and delete the var, the port goes along happily hanging onto a reference to that var, so you won't be able to easily add the var to anything else, but it still "works" fine.

Reply:
So do we just rely on the user to clean up references for stuff they delete? If so I can get rid of the deconstruct thing.